### PR TITLE
payload config: add new Probability<const u32: MIN_AS_BITS> type

### PR DESCRIPTION
**Pull request by goxberry**
### What does this PR do?

Adds a new `Probability<const u32: MIN_AS_BITS>` type as a proof-of-concept for configuration fields that accept `f32` values interpreted as probabilities. The type isn't used yet. The `MIN_AS_BITS` field is intended for use as a lower bound in cases where the minimum probability for a configuration field should be positive, e.g., unique tag probability of at least 0.1.

### Motivation

Reduce some boilerplate validation code related to probability-valued configuration fields for payloads.

### Related issues

n/a

### Additional Notes

Tests and ergonomics are still a work in progress. The plan files will be removed prior to merging the PR. These files are present in case reviewers find them helpful, but I don't intend them to clutter the repo.

### CODE REVIEW

**Comment by goxberry** [comment]
* **#1874** <a href="https://app.graphite.com/github/pr/DataDog/lading/1874?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/DataDog/lading/1874?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: docs/plans/probability-property-tests.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+3. JSON and YAML round trips rely on Ryu `f32` printing being round-trip exact
+   for finite values — true for both `serde_json` and `serde_yaml`.
+
+## Outcome
+
+Approximately 18 unit tests collapse into 12 generic property tests
+(of which the `MIN`-parameterized ones expand to three `#[test]` functions
+each) plus 6 retained unit tests. Input-space coverage broadens substantially
+while the literal wire formats and discrete edge cases (`-0.0`, `[1.0, 1.0]`)
+remain pinned by unit tests.
```
Comment (blt) [inline docs/plans/probability-property-tests.md]: Neat.

**Review by blt — APPROVED** [review]

File: lading_payload/src/common/config.rs [hunk 2]
```diff
@@ -84,3 +84,309 @@ where
...
 }
+
+/// Bit pattern of `-0.0_f32`. Used to reject negative zero, which compares
+/// equal to `+0.0` under IEEE-754 numeric ordering.
+const NEG_ZERO_AS_BITS: u32 = 0x8000_0000;
+
+/// Error returned when a value cannot be turned into a [`Probability`].
+#[derive(Debug, thiserror::Error, Clone, Copy)]
+pub enum ProbabilityError {
+    /// Value is NaN or `±∞`.
```
[the remark this answers is not in the document]
Comment (goxberry) [reply lading_payload/src/common/config.rs]: Changed in https://github.com/DataDog/lading/pull/1874/commits/1a6683dd0d3365b084aa89755e4f9b33e3adb1d7.

**Comment by goxberry** [comment]
Will submit a follow-on PR with a few more ergonomics tweaks, and wire up payload configs to see whether this experiment proves out.
